multi_point.go

```diff
@@ -1,7 +1,15 @@
 package geom
 
+// MultiPoint is a geometry with multiple points.
 type MultiPoint [][2]float64
 
-func (mp MultiPoint) Points() [][2]float64 {
-	return mp
+// Points returns the coordinates for the points
+func (mp *MultiPoint) Points() [][2]float64 {
+	return *mp
```

Comment:
No need for the pointer here. If you are going to use a pointer you need to make sure it's not nil, before deref'ing it.

Reply:
My mistake, I didn't realize you could mix/match pointer receivers and non-pointer receivers on the same struct and still have it implement a particular interface.

Reply:
No worries. I had that confusion when I was first working with Go as well. I still find pointers to be bit tricky in go sometimes. Take a bit to get used to.